hack/registry-cache-image-overwrite.sh

```diff
@@ -0,0 +1,20 @@
+# SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+repository=$(echo $SKAFFOLD_IMAGE | rev | cut -d':' -f 2- | rev)
+tag=$(echo $SKAFFOLD_IMAGE | rev | cut -d':' -f 1 | rev)
+
+cat <<EOF > local-setup/patch-controller-deployment-image.yaml
```

Comment:
Nit: To have a consistent naming with the above suggestion

```suggestion
cat <<EOF > local-setup/patch-controllerdeployment-image.yaml
```